api/src/test/java/com/epam/pipeline/acl/region/CloudRegionApiServiceTest.java

```diff
@@ -0,0 +1,231 @@
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldUpdateAbstractCloudRegionForAdmin() {
+        doReturn(region).when(mockCloudRegionManager).update(region.getId(), cloudRegionDTO);
+
+        assertThat(cloudRegionApiService.update(region.getId(), cloudRegionDTO)).isEqualTo(region);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldNotUpdateAbstractCloudRegionForNotAdmin() {
+        doReturn(region).when(mockCloudRegionManager).update(region.getId(), cloudRegionDTO);
+
+        cloudRegionApiService.update(region.getId(), cloudRegionDTO);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldDeleteAbstractCloudRegionForAdmin() {
+        doReturn(region).when(mockCloudRegionManager).delete(region.getId());
+
+        assertThat(cloudRegionApiService.delete(region.getId())).isEqualTo(region);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldDeleteAbstractCloudRegionForNotAdmin() {
+        doReturn(region).when(mockCloudRegionManager).delete(region.getId());
+
+        cloudRegionApiService.delete(region.getId());
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadAllAvailableCloudsForAdmin() {
+        doReturn(availableCloudsList).when(mockCloudRegionManager).loadAllAvailable(CloudProvider.AWS);
+
+        assertThat(cloudRegionApiService.loadAllAvailable(CloudProvider.AWS)).isEqualTo(availableCloudsList);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    @WithMockUser(roles = SIMPLE_USER_ROLE)
+    public void shouldNotLoadAllAvailableCloudsForNotAdmin() {
+        doReturn(availableCloudsList).when(mockCloudRegionManager).loadAllAvailable(CloudProvider.AWS);
+
+        cloudRegionApiService.loadAllAvailable(CloudProvider.AWS);
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldReturnAllCloudRegionsForAdmin() {
+        doReturn(clouds).when(mockCloudRegionManager).loadAll();
+
+        final List<? extends AbstractCloudRegion> abstractCloudRegions = cloudRegionApiService.loadAll();
+
+        assertThat(abstractCloudRegions).isEqualTo(clouds);
+        assertThat(abstractCloudRegions.get(0).getId()).isEqualTo(clouds.get(0).getId());
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER_ROLE)
+    public void shouldReturnAllCloudRegionsWhenPermissionIsGranted() {
+        initAclEntity(region,
+                Collections.singletonList(new UserPermission(SIMPLE_USER_ROLE, AclPermission.READ.getMask())));
+        doReturn(clouds).when(mockCloudRegionManager).loadAll();
+
+        final List<? extends AbstractCloudRegion> abstractCloudRegions = cloudRegionApiService.loadAll();
+
+        assertThat(abstractCloudRegions).isEqualTo(clouds);
+        assertThat(abstractCloudRegions.get(0).getId()).isEqualTo(clouds.get(0).getId());
```

Comment:
Lets make a new expected object that will not be mutated during the test execution

Reply:
Done. I hope that i understand you correctly.